src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -158,6 +158,12 @@ public function should_update( $type, $item, $context ) {
 		// Next up, is this an item we can update?
 		if ( 'core' === $type ) {
 			$update = Core_Upgrader::should_update_to_version( $item->current );
+		} elseif ( 'plugin' === $type || 'theme' === $type ) {
+			$update = false;
+			if ( wp_is_auto_update_enabled_for_type( $type ) ) {
+				$auto_updates = (array) get_site_option( "auto_update_{$type}s", array() );
+				$update      = in_array( $item->{$type}, $auto_updates, true ) || ! empty( $item->autoupdate );
+			}
```

Comment:
As far as I can tell, this will stop autoupdates being triggered from the API if the corresponding UI filter is disabled. I think it should be like this where we default to the value of the API response.

```suggestion
			$update =  ! empty( $item->autoupdate );
			if ( ! $update && wp_is_auto_update_enabled_for_type( $type ) ) {
				$auto_updates = (array) get_site_option( "auto_update_{$type}s", array() );
				$update      = in_array( $item->{$type}, $auto_updates, true ;
			}
```

Reply:
>this will stop autoupdates being triggered from the API if the corresponding UI filter is disabled

That is the intention and the way the feature plugin has worked all along.

Maybe that is wrong, but here's the reasoning behind it (as I see it):

1. if the UI has been disabled, then the site owner/admin has explicitly said they do not want **any** auto-updates for plugins/themes (even "forced security pushes")
2. if the UI is enabled, then it defaults to the value of the API response

Reply:
> if the UI has been disabled, then the site owner/admin has explicitly said they do not want any auto-updates for plugins/themes (even "forced security pushes")

I disagree with this premise, that is the purpose of the `AUTOMATIC_UPDATER_DISABLED` constant. 

I think this will lead to less secure sites when users turn off the auto updates UI because they don't want to or don't want their clients to start auto-updating plugins, and then .org won't be able to push out emergency security updates for a number of sites.

Reply:
Looking at the code, I also think that isn't how the feature plugin works.

https://github.com/WordPress/wp-autoupdates/blob/5a7cb834b3e73d9829cd62b0d1386350d9b7f535/functions.php#L204

If the UI is disabled, it passes the original auto update value which is the `autoupdate` value from the API.

It looks like it also stops auto updates if the plugin doesn't have auto updates enabled, though that has been fixed in this PR.

Reply:
Thank you for catching this!

1. your change is what should be in core (I kind of reverse engineered the rationale I gave to fit the incorrect patch)
2. what is in the plugin is not quite correct, it should be:

```
function wp_autoupdates_selected_plugins( $update, $item ) {
	if ( wp_autoupdates_is_plugins_auto_update_enabled() ) {
		$auto_updates = (array) get_site_option( 'wp_auto_update_plugins', array() );
		return in_array( $item->plugin, $auto_updates, true ) || $update;
	}

	return $update;
}
```

Reply:
That looks great to me!

Reply:
I actually wrote the above comment several hours but forgot to submit it :-)

I'll do a PR for the plugin with that change.